CIP-0026/CIP-0026.md

```diff
@@ -0,0 +1,468 @@
+
+## Security
+
+Most of the security considerations relate to non-verifiable metadata. Verifiable metadata can generally always be accepted as is (provided that it is verified). Our threat model is that non-verifiable metadata may always have been provided by an attacker.
+
+Accepting non-verifiable metadata blindly can lead to attacks. For example, a malicious server or user might attempt to name their currency “Ada”. If we blindly accept this and overwrite the existing mapping for “Ada”, this would lead to easy phishing attacks. The approach we take is heavily inspired by petname systems, GPG keyservers, and local address books. The user always has a local mapping which is trusted, and then adding to or updating that mapping requires explicit user consent, unless we can prove that this is trustworthy (i.e. the metadata is verifiable).
+
+How can users decide whether to trust an update? This is where the attestation signatures come into play. If a user trusts the entity which signs the metadata record, that may be sufficient for them to accept it as a legitimate update. See the earlier “Trust” sections for some discussion of trusted keys and updates.
+
+Clients could also be tricked into downloading large amounts of metadata that the user does not want. For this reason clients should expose some kind of configurable download limiting, and we suggest that the server set the Content-Length header to support this. However, this problem is no worse than that faced by the average web browser, so we do not think it will be a problem in practice.
+
+Clients also need to worry about replay attacks, where they are sent old (correctly signed) records in an attempt to “roll back” a legitimate update. The easiest way to avoid this is to correctly implement sequence numbers, in which case old updates will be rejected.
+
+## Client/consumer split
+
+This document features a perhaps surprisingly large number of components. Some of these are uncontroversial: there must be a server and a client, and you need an agreed format to communicate.
+
+But the client/consumer split is less clear. Strictly, it is not necessary, and the two could be treated as one component. But splitting them allows the client to be a simpler, self-contained component, while the consumer may have concerns that reach into the details of the application that wants to use the metadata.
+
+## Consumer sanity checking
+
+Keeping most of the sanity checking (e.g. that names are unique) in the consumer simplifies the behaviour of the client and server significantly, and allows different consumers to adopt different policies as appropriate, even if they share a client.
+
+## Data storage location
+
+This design needs to store a fair amount of data in a shared location. We might wonder whether we should use the blockchain for this: we could store metadata updates in transaction metadata.
```

Comment:
This section is partially my argument against storing any metadata mappings on-chain... which is precisely something we then (sort-of) endorse earlier in the document. My preferred resolution to that tension is to delete the previous bit about storing things on-chain, of course :)

Reply:
Note that, the suggestion above isn't about storing the metadata on-chain, but simply a "link" between some keys and some metadata **subject**. The metadata would still be stored off-chain in this setup, but the chain used as a medium to convey trusted keys.